src/lib/traversals.js

```diff
@@ -0,0 +1,55 @@
+'use strict';
+
+// Time O(N), where N is number of nodes in the tree
+// Space O(H) where H is the height of the tree
+const preOrderTraversal = (rootNode, treeValues) => {
+  if (!rootNode) {
+    return undefined;
+  }
+  treeValues.push(rootNode.value);
+  preOrderTraversal(rootNode.left, treeValues);
+  preOrderTraversal(rootNode.right, treeValues);
+  return undefined;
+};
+
+// Time O(N), where N is number of nodes in the tree
+// Space O(H) where H is the height of the tree
+const postOrderTraversal = (rootNode, treeValues) => {
+  if (!rootNode) {
+    return undefined;
+  }
+  postOrderTraversal(rootNode.left, treeValues);
+  postOrderTraversal(rootNode.right, treeValues);
+  treeValues.push(rootNode.value);
+  return undefined;
+};
+
+// Time O(N), where N is number of nodes in the tree
+// Space O(H) where H is the height of the tree
+const inOrderTraversal = (rootNode, treeValues) => {
+  if (!rootNode) {
+    return undefined;
+  }
+  inOrderTraversal(rootNode.left, treeValues);
+  treeValues.push(rootNode.value);
+  inOrderTraversal(rootNode.right, treeValues);
+
+  return undefined;
+};
+
+// Time O(N),
+// Space O(N), where N is number of nodes in the tree
```

Comment:
Actually, space is O(h) where `h` is the height of tree, i.e. number of edges from the root to to the lowest node.